packages/plugin-ext/src/plugin/languages/lens.ts

```diff
@@ -0,0 +1,82 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import URI from 'vscode-uri/lib/umd';
+import * as theia from '@theia/plugin';
+import { DocumentsExtImpl } from '../documents';
+import { CodeLensSymbol } from '../../api/model';
+import * as Converter from '../type-converters';
+import { ObjectIdentifier } from '../../common/object-identifier';
+import { createToken } from '../token-provider';
+import { CommandsConverter } from '../command-registry';
+
+export class CodeLensAdapter {
```

Comment:
maybe explain a little what is doing adapter ?

Reply:
added jsdoc